src/Avalonia.Base/Threading/AvaloniaSynchronizationContext.cs

```diff
@@ -61,17 +61,35 @@ public static void InstallIfNeeded()
         /// <inheritdoc/>
         public override void Post(SendOrPostCallback d, object? state)
         {
-            _dispatcher.Post(d, state, Priority);
+            try
+            {
+                _dispatcher.Post(d, state, Priority);
+            }
+            catch (Exception e)
```

Comment:
This does not look right, `Post` does not execute the callback, it only adds it to the queue.